pkg/syncer/spec/mutators/deployment.go

```diff
@@ -39,9 +51,12 @@ func (dm *DeploymentMutator) GVR() schema.GroupVersionResource {
 	}
 }
 
-func NewDeploymentMutator(upstreamURL *url.URL) *DeploymentMutator {
+func NewDeploymentMutator(upstreamURL *url.URL, secretLister cache.Indexer, clusterName logicalcluster.Name, indexName string) *DeploymentMutator {
 	return &DeploymentMutator{
-		upstreamURL: upstreamURL,
+		clusterName:  clusterName,
```

Comment:
why store a clusterName. The downstream object has that info, doesn't it?

Reply:
it's even wrong. There is no one clusterName anymore. The syncer speaks with the vw and that has many workspaces.